lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -288,21 +293,30 @@ def create_cluster(module, redshift):
               'number_of_nodes', 'publicly_accessible',
               'encrypted', 'elastic_ip', 'enhanced_vpc_routing'):
         if p in module.params:
-            params[p] = module.params.get(p)
+            # https://github.com/boto/boto3/issues/400
+            if module.params.get(p) is not None:
+                params[p] = module.params.get(p)
+
+    if d_b_name:
+        params[d_b_name] = d_b_name
 
     try:
-        redshift.describe_clusters(identifier)['DescribeClustersResponse']['DescribeClustersResult']['Clusters'][0]
+        redshift.describe_clusters(ClusterIdentifier=identifier)['Clusters'][0]
         changed = False
-    except boto.exception.JSONResponseError as e:
+    except botocore.exceptions.ClientError as e:
```

Comment:
This should catch only the specific error that occurs when the cluster doesn't exist yet. You can import is_boto3_error_code from module_utils.aws.core and do: `except is_boto3_error_code('ClusterNotFound'):`